src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,316 @@
+    /// This is only offered on C# 11 and above where this delegate can be cached and will not cause allocations each
+    /// time.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.RemoveUnnecessaryLambdaExpressionDiagnosticId,
+                   EnforceOnBuildValues.RemoveUnnecessaryLambdaExpression,
+                   CSharpCodeStyleOptions.PreferMethodGroupConversion,
+                   LanguageNames.CSharp,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Remove_unnecessary_lambda_expression), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Lambda_expression_can_be_removed), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                   isUnnecessary: true)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.LanguageVersion().IsCSharp11OrAbove())
+                {
+                    var expressionType = context.Compilation.ExpressionOfTType();
+                    context.RegisterSyntaxNodeAction(
+                        c => AnalyzeSyntax(c, expressionType),
+                        SyntaxKind.SimpleLambdaExpression, SyntaxKind.ParenthesizedLambdaExpression, SyntaxKind.AnonymousMethodExpression);
+                }
+            });
+        }
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol? expressionType)
+        {
+            var cancellationToken = context.CancellationToken;
+            var semanticModel = context.SemanticModel;
+            var syntaxTree = semanticModel.SyntaxTree;
+
+            var preference = context.GetOption(CSharpCodeStyleOptions.PreferMethodGroupConversion);
+            if (preference.Notification.Severity == ReportDiagnostic.Suppress)
+            {
+                // User doesn't care about this rule.
+                return;
+            }
+
+            var anonymousFunction = (AnonymousFunctionExpressionSyntax)context.Node;
+
+            // Syntax checks first.
+
+            // Don't simplify static lambdas.  The user made them explicitly static to make it clear it doesn't capture
+            // anything.  If we get rid of the lambda it won't be clear anymore if capturing/allocs are happening or not.
```

Comment:
Not sure I understand this.. converting to a method group can't possible capture, right, so how would it be a detriment? 

Or is this more to prevent someone coming back and making it a non-static lambda in a subsequent change?

Reply:
So with a static lambda, you get a single alloc (the cached delegate). With a member group conversation you may get a single cached alloc, or an alloc each time. 

There's no way with a member group to ensure you're in the first category. 

You're right that is not Amy the capture though. I'll update the docs.